lib/node.js

```diff
@@ -37,12 +49,12 @@ Node.prototype.path = function (i) {
 Node.prototype.final = function () {
   if (!this.valueBuffer || this.value !== null) return this
   this.value = this.valueEncoding ? this.valueEncoding.decode(this.valueBuffer) : this.valueBuffer
+  this.flags = this.flags >> 8
```

Comment:
needs a comment

Reply:
Added a comment explaining that this is required to support user-defined flags while hiding internal flags.